scripts/kube-init.sh

```diff
@@ -54,48 +59,51 @@ wget -O kubectl "http://storage.googleapis.com/kubernetes-release/release/${K8S_
 sudo chmod +x kubectl
 sudo mv kubectl /usr/local/bin/
 
-echo "Download localkube from minikube project"
-wget -O localkube "https://storage.googleapis.com/minikube/k8sReleases/v1.7.0/localkube-linux-amd64"
-sudo chmod +x localkube
-sudo mv localkube /usr/local/bin/
-
-echo "Starting localkube"
-sudo nohup localkube --logtostderr=true --enable-dns=false > localkube.log 2>&1 &
-
-echo "Waiting for localkube to start..."
-if ! timeout 120 sh -c "while ! curl -ks http://127.0.0.1:8080/ >/dev/null; do sleep 1; done"; then
-    sudo cat localkube.log
-    die $LINENO "localkube did not start"
+echo "Download minikube from minikube project"
+wget -O minikube "https://storage.googleapis.com/minikube/releases/v0.30.0/minikube-linux-amd64"
+sudo chmod +x minikube
+sudo mv minikube /usr/local/bin/
+
+# L68-100: Set up minikube within Travis CI
+# See https://github.com/kubernetes/minikube/blob/master/README.md#linux-continuous-integration-without-vm-support
+echo "Set up minikube"
+export MINIKUBE_WANTUPDATENOTIFICATION=false
+export MINIKUBE_WANTREPORTERRORPROMPT=false
+export CHANGE_MINIKUBE_NONE_USER=true
+sudo mkdir -p $HOME/.kube || true
+sudo mkdir -p $HOME/.minikube || true
+sudo touch $HOME/.kube/config
+export KUBECONFIG=$HOME/.kube/config
+export MINIKUBE_HOME=$HOME
+export MINIKUBE_DRIVER=${MINIKUBE_DRIVER:-none}
+
+# Used bootstrapper to be kubeadm for the most recent k8s version
+# since localkube is depreciated and only supported up to version 1.10.0
+echo "Starting minikube"
+sudo minikube start --vm-driver=$MINIKUBE_DRIVER --bootstrapper=kubeadm --kubernetes-version=$K8S_VERSION --logtostderr
+
+# Make sure minikube information is not outdated
+sudo minikube update-context
```

Comment:
Why do we need to run `minikube update-context`? I assume it modifies the kubeconfig file to point to the minikube cluster and I'd expect `minikube start` does that for us already

Reply:
It's effectively a safeguard: https://github.com/kubernetes/minikube/blob/master/cmd/minikube/cmd/update-context.go#L36